crates/storage/src/connection/block.rs

```diff
@@ -179,6 +179,27 @@ impl Transaction<'_> {
             )
             .context("Deleting block from trie_storage_removals table")?;
 
+        self.inner()
+            .execute(
+                "DELETE FROM trie_class WHERE block_number = ?",
+                params![&block],
+            )
+            .context("Deleting block from trie_class table")?;
+
+        self.inner()
+            .execute(
+                "DELETE FROM trie_contracts WHERE block_number = ?",
```

Comment:
There's no `block_number` in `trie_contracts`.

Ideally we'd be using the same trie code here that takes care of removing trie nodes first and then removes the indices (`prune_trie()`).

Reply:
No, you're fundamentally right.

However, we have no way of knowing which nodes were added at that block. And even if we did we cannot just remove those nodes without making further adjustments (that's why we have state revert). We _do_ have the indices of nodes that are guaranteed to be inaccessible at that block, so we should just remove at least those before we completely lose track of that information.

How we actually handle the revert is another good question: I think state reverts might be broken now as nodes removed during the revert should be removed right away and not delayed. That's not something to be handled in this PR though.